packages/babel-parser/src/parser/statement.js

```diff
@@ -1022,6 +1031,22 @@ export default class StatementParser extends ExpressionParser {
     );
   }
 
+  checkVarIdHasLet(id: ?N.Pattern): ?N.Identifier {
```

Comment:
I wonder if this would be clearer if it returned a boolean. Maybe something like `isValidVarName`?

Reply:
We need the identifier to report the error location.

Reply:
I guess it could return a tuple, but then I don’t know if that’s clearer or not.